docs/template_plugin/tests/functional/op_reference/einsum.cpp

```diff
@@ -0,0 +1,184 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <gtest/gtest.h>
+
+#include "openvino/opsets/opset7.hpp"
+#include "openvino/opsets/opset1.hpp"
+#include "base_reference_test.hpp"
+
+using namespace reference_tests;
+using namespace ov;
+
+namespace {
+struct EinsumParams {
+    std::vector<Tensor> inputs;
+    std::string equation;
+    Tensor expectedResult;
+    std::string testcaseName;
+};
+
+struct Builder : ParamsBuilder<EinsumParams> {
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, inputs);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, equation);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, expectedResult);
+    REFERENCE_TESTS_ADD_SET_PARAM(Builder, testcaseName);
+};
+
+class ReferenceEinsumTest : public testing::TestWithParam<EinsumParams>, public CommonReferenceTest {
+public:
+    void SetUp() override {
+        auto params = GetParam();
+        function = CreateFunction(params);
+        for (const auto& input_tensor : params.inputs) {
+            inputData.push_back(input_tensor.data);
+        }
+        refOutData = {params.expectedResult.data};
+    }
+
+    static std::string getTestCaseName(const testing::TestParamInfo<EinsumParams>& obj) {
+        auto param = obj.param;
+        std::ostringstream result;
+        result << "iType=" << param.inputs[0].type;
+        result << "_iShape=" << param.inputs[0].shape;
+        result << "_equation=" << param.equation;
+        result << "_eType=" << param.expectedResult.type;
+        if (param.testcaseName != "") {
```

Comment:
How about that?

result << "_eShape=" << param.expectedResult.shape;
if (!param.testcaseName.empty()) {       
            result << "_=" << param.testcaseName;
}

Reply:
Updated, thanks for your comment